drivers/storage/portworx/components_test.go

```diff
@@ -5830,6 +5831,130 @@ func TestCSI_0_3_NodeAffinityChange(t *testing.T) {
+					Enabled: true,
+					AlertManager: &corev1.AlertManagerSpec{
+						Enabled: true,
+					},
+				},
+			},
+		},
+		Status: corev1.StorageClusterStatus{
+			DesiredImages: &corev1.ComponentImages{
+				PrometheusOperator: manifest.DefaultPrometheusOperatorImage,
+			},
+		},
+	}
+
+	// Setup manifest and config directory
+	m := &fakeManifest{}
+	m.Init(nil, nil, k8sutil.K8sVer1_22)
+	manifest.SetInstance(m)
+	defer manifestSetup()
+	pxutil.SpecsBaseDir = func() string {
+		return "../../../bin/configs"
+	}
+	defer func() {
+		pxutil.SpecsBaseDir = func() string {
+			return pxutil.PortworxSpecsDir
+		}
+	}()
+
+	// TestCase: k8s 1.22 should upgrade Prometheus operator image
+	driver.SetDefaultsOnStorageCluster(cluster)
+	cluster.Spec.Placement = nil
+	require.Equal(t, "quay.io/prometheus-operator/prometheus-operator:v1.2.3", cluster.Status.DesiredImages.PrometheusOperator)
+
+	k8sClient.Create(
+		context.TODO(),
+		&v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      component.AlertManagerConfigSecretName,
+				Namespace: cluster.Namespace,
+			},
+		},
+		&client.CreateOptions{},
+	)
+
+	err := driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Prometheus operator ClusterRole should be updated as well
+	expectedCR := testutil.GetExpectedClusterRole(t, "prometheusOperatorClusterRole.yaml")
+	actualCR := &rbacv1.ClusterRole{}
+	err = testutil.Get(k8sClient, actualCR, component.PrometheusOperatorClusterRoleName, "")
+	require.NoError(t, err)
+	require.Equal(t, expectedCR.Name, actualCR.Name)
+	require.Empty(t, actualCR.OwnerReferences)
+	require.ElementsMatch(t, expectedCR.Rules, actualCR.Rules)
+
+	// Prometheus operator Deployment
+	expectedDeployment := testutil.GetExpectedDeployment(t, "prometheusOperatorDeployment_v0.50.0.yaml")
+	deployment := &appsv1.Deployment{}
+	err = testutil.Get(k8sClient, deployment, component.PrometheusOperatorDeploymentName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, expectedDeployment.Name, deployment.Name)
+	require.Equal(t, expectedDeployment.Namespace, deployment.Namespace)
+	require.Len(t, deployment.OwnerReferences, 1)
+	require.Equal(t, cluster.Name, deployment.OwnerReferences[0].Name)
+	require.Equal(t, expectedDeployment.Labels, deployment.Labels)
+	require.Equal(t, expectedDeployment.Annotations, deployment.Annotations)
+	require.Equal(t, expectedDeployment.Spec, deployment.Spec)
+
+	// Prometheus CRDs should be registered
+	crdNameList := []string{
+		"alertmanagerconfigs.monitoring.coreos.com",
+		"alertmanagers.monitoring.coreos.com",
+		"podmonitors.monitoring.coreos.com",
+		"probes.monitoring.coreos.com",
+		"prometheuses.monitoring.coreos.com",
+		"prometheusrules.monitoring.coreos.com",
+		"servicemonitors.monitoring.coreos.com",
+		"thanosrulers.monitoring.coreos.com",
+	}
+	go func() {
+		for _, crdName := range crdNameList {
+			err := testutil.ActivateCRDWhenCreated(fakeExtClient, crdName)
+			require.NoError(t, err)
+		}
+	}()
```

Comment:
nit: I guess we might not need this. Just try and decide whether to remove this or not.